File: 1518.water-bottles.py
#
# @lc app=leetcode id=1518 lang=python3
#
# [1518] Water Bottles
#

# @lc code=start
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = numBottles
        empty = numBottles

        while empty >= numExchange:
            exchanged = empty // numExchange
            empty %= numExchange
            res += exchanged
            empty += exchanged
        
        return res
        
# @lc code=end

File: 1518.water-bottles.java
/*
 * @lc app=leetcode id=1518 lang=java
 *
 * [1518] Water Bottles
 */

// @lc code=start
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int empty, res, exchanged;
        empty = res = numBottles;

        while (empty >= numExchange){
            exchanged = Math.floorDiv(empty, numExchange);
            res += exchanged;
            empty = exchanged + (empty % numExchange);
        };

        return res;
    }
}
// @lc code=end

